### PR TITLE
Adds base doc for the core

### PR DESCRIPTION
**Pull request by alanzanattadev**

### CODE REVIEW

File: docs/core-execution.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Molecule has several internal concepts to represent the different parts of the whole architecture. Among them there are:
+
+- Strategy
+- Runner
+- Stager
+- Controller
+- Plan
+- Execution
+- Plugin
+- Core
```
Comment (GauthierPLM) [inline docs/core-execution.md]: - Add links to each item to jump to the corresponding part of the doc
- Order them as they are explained in the doc

File: docs/core-execution.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
+---
+title: Tool execution
+---
+
+## Concepts
+
+Molecule has several internal concepts to represent the different parts of the whole architecture. Among them there are:
```
Comment (GauthierPLM) [inline docs/core-execution.md]: Miss a part that link to the rest of the content.
`This document will introduce you to these concepts and will help you to understand the internal working of Molecule.`

File: docs/core-execution.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+The code related to strategies is in ExecutionControlEpic/LanguageServerProtocolFeature/Model/ 
+
+### Runner
+
+Once the strategy is generated, it is interpreted by what we call a runner. Each strategy has a type (a word describing the method) and each type is associated with a specific runner. The runner is called to run or stop a strategy.
+
+It's basically a generic interface to make easy the addition of new ways of executing tools in Molecule. It provides a unified API for any kind of tool execution.
+
+Because Molecule supports language servers (with the Language Server Protocol, LSP), each runner has the ability to define whether it supports it or not. Indeed, some strategy involves unreadable outputs (like the pty strategy which makes use of ansii codes) leading to the inability to parse the LSP. It's the role of the runner to tell this information to Molecule.
```
Comment (GauthierPLM) [inline docs/core-execution.md]: It would be nice to add a link to the LSP doc.

File: docs/core-execution.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+- to run the controller
+- to provide an input stream, an output stream and an event emitter for specific events (controller killed, exit, etc ...).
+
+The code related to stagers is in ExecutionControlEpic/LanguageServerProtocolFeature/Model/ 
+
+### Core
+
+#### Architecture
+
+Molecule's Core uses Redux and the flux pattern. If you don't know this architecture I let you check here: https://facebook.github.io/flux/docs/in-depth-overview.html#content
```
Comment (GauthierPLM) [inline docs/core-execution.md]: To discover the Flux pattern, you can[ read its official documentation](https://facebook.github.io/flux/docs/in-depth-overview.html#content).

File: docs/core-execution.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+In order to handle UI changes and input controls in a consistent way, we chose this architecture. All the molecule state is stored in the Redux state, and state changes lead to a UI refresh. Modifications to the state is made by reducers which receive actions. Actions are simple descriptions of events happening.
+
+However, Molecule being a highly asynchronous environment, we chose to use redux-observable which allows us to work with RxJS (and so observables) to handle asynchronous parts.
+
+Actions dispatched can trigger asynchronous behaviors, that dispatch themselves new actions. This handler which uses actions to dispatch other actions during asynchronous behaviors is called an Epic.
+
+#### Epics
+
+Epics are described here https://redux-observable.js.org/docs/basics/Epics.html.
```
Comment (GauthierPLM) [inline docs/core-execution.md]: Epics are described in [Redux-Observable documentation](https://redux-observable.js.org/docs/basics/Epics.html).

**Review by GauthierPLM — CHANGES_REQUESTED** [review]
The writing is good but adding schema (for example detailling the different layers of a plugin) and examples (a plugin example) would help a lot in the understanding.

File: docs/core-workflow.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+- Layouts: which contains some old visual components
+- AtomLinks: which contains the functions that mount react components (ReactDOM.render) on Atom elements.
+- Model: which contains core functions for data management
+- Process: which contains files being launched as child processes
+
+## Code & Debug
+
+Let's say you want to add a new feature and that this feature contains a User Interface.
+
+### Storybook
```
Comment (GauthierPLM) [inline docs/core-workflow.md:53]: We no longer use Storybook in Molecule as our integration was outdated and broken.

Comment (alanzanattadev) [reply docs/core-workflow.md:53]: We have to make it work again, it's a really useful tool

**Review by GauthierPLM — APPROVED** [review]

**Pull request merged**